tests/integrated/test-drift-instability/2fluid.cxx

```diff
@@ -3,415 +3,420 @@
-Field2D Rxy, Bpxy, Btxy, hthe;
-
-// parameters
-BoutReal Te_x, Ti_x, Ni_x, Vi_x, bmag, rho_s, fmei, AA, ZZ;
-BoutReal lambda_ei, lambda_ii;
-BoutReal nu_hat, mui_hat, wci, nueix, nuiix;
-BoutReal beta_p;
-
-// settings
-bool estatic, ZeroElMass; // Switch for electrostatic operation (true = no Apar)
-BoutReal zeff, nu_perp;
-bool evolve_rho, evolve_te, evolve_ni, evolve_ajpar, evolve_vi, evolve_ti;
-BoutReal ShearFactor;
-
-// Inversion objects
-std::unique_ptr<Laplacian> phi_solver;
-std::unique_ptr<Laplacian> apar_solver;
-
-FieldGroup comms; // Group of variables for communications
-
-Coordinates *coord; // Coordinate system
-
-CELL_LOC maybe_ylow;
-
-int physics_init(bool UNUSED(restarting)) {
-  Field2D I; // Shear factor 
-  
-  output.write("Solving 6-variable 2-fluid equations\n");
-  
-  /************* LOAD DATA FROM GRID FILE ****************/
-
-  // Load 2D profiles (set to zero if not found)
-  GRID_LOAD(Ni0);
-  GRID_LOAD(Ti0);
-  GRID_LOAD(Te0);
-  GRID_LOAD(Vi0);
-  GRID_LOAD(Ve0);
-  GRID_LOAD(phi0);
-  GRID_LOAD(rho0);
-  GRID_LOAD(Ajpar0);
-
-  // Load magnetic curvature term
-  b0xcv.covariant = false; // Read contravariant components
-  mesh->get(b0xcv, "bxcv"); // b0xkappa terms
-
-  // Coordinate system
-  coord = mesh->getCoordinates();
-
-  // Load metrics
-  GRID_LOAD(Rxy);
-  GRID_LOAD(Bpxy);
-  GRID_LOAD(Btxy);
-  GRID_LOAD(hthe);
-  mesh->get(coord->dx,   "dpsi");
-  mesh->get(I,    "sinty");
-
-  // Load normalisation values
-  GRID_LOAD(Te_x);
-  GRID_LOAD(Ti_x);
-  GRID_LOAD(Ni_x);
-  GRID_LOAD(bmag);
-
-  Ni_x *= 1.0e14;
-  bmag *= 1.0e4;
-
-  /*************** READ OPTIONS *************************/
-
-  // Read some parameters
-  Options *globalOptions = Options::getRoot();
-  Options *options = globalOptions->getSection("2fluid");
-  OPTION(options, AA, 2.0);
-  OPTION(options, ZZ, 1.0);
-
-  OPTION(options, estatic,     false);
-  OPTION(options, ZeroElMass,  false);
-  OPTION(options, zeff,        1.0);
-  OPTION(options, nu_perp,     0.0);
-  OPTION(options, ShearFactor, 1.0);
-  
-  (globalOptions->getSection("Ni"))->get("evolve", evolve_ni,    true);
-  (globalOptions->getSection("rho"))->get("evolve", evolve_rho,   true);
-  (globalOptions->getSection("vi"))->get("evolve", evolve_vi,   true);
-  (globalOptions->getSection("te"))->get("evolve", evolve_te,   true);
-  (globalOptions->getSection("ti"))->get("evolve", evolve_ti,   true);
-  (globalOptions->getSection("Ajpar"))->get("evolve", evolve_ajpar, true);
-  
-  if(ZeroElMass)
-    evolve_ajpar = false; // Don't need ajpar - calculated from ohm's law
-
-  /*************** INITIALIZE LAPLACIAN SOLVERS ********/
-  phi_solver = Laplacian::create(globalOptions->getSection("phisolver"));
-  if (!estatic && !ZeroElMass) {
-    apar_solver = Laplacian::create(globalOptions->getSection("aparsolver"));
-  }
+// just define a macro for V_E dot Grad
+#define vE_Grad(f, p) (b0xGrad_dot_Grad(p, f) / coord->Bxy)
+
+class TwoFluid : public PhysicsModel {
```

Comment:
warning: constructor does not initialize these fields: Te_x, Ti_x, Ni_x, Vi_x, bmag, rho_s, fmei, AA, ZZ, lambda_ei, lambda_ii, nu_hat, mui_hat, wci, nueix, nuiix, beta_p, estatic, ZeroElMass, zeff, nu_perp, evolve_rho, evolve_te, evolve_ni, evolve_ajpar, evolve_vi, evolve_ti, ShearFactor, coord, maybe_ylow [cppcoreguidelines-pro-type-member-init]

```cpp
class TwoFluid : public PhysicsModel {
      ^
```